src/hotspot/share/opto/vectornode.cpp

```diff
@@ -150,9 +150,21 @@ int VectorNode::opcode(int sopc, BasicType bt) {
   case Op_RoundDoubleMode:
     return (bt == T_DOUBLE ? Op_RoundDoubleModeV : 0);
   case Op_RotateLeft:
-    return (is_integral_type(bt) ? Op_RotateLeftV : 0);
+    switch (bt) {
+    case T_INT:
+    case T_LONG: return Op_RotateLeftV;
+    default:     return 0; // RotateLeftV for byte, short values produces incorrect Java result.
+                           // Because java code should convert a byte, short value into int value,
```

Comment:
Current handling creates a vector rotate IR nodes for all integral types and later on dismantles it into constituent operation (SHIFTs and ORs).  Idea behind this to intrinsify lanewise vector operation for any integral type.

Your above change will prevent intrinsification and test may still pass.